examples/pnet/index.js

```diff
@@ -0,0 +1,48 @@
+/* eslint no-console: ["off"] */
+'use strict'
+
+const { generate } = require('libp2p/src/pnet')
+const privateLibp2pNode = require('./libp2p-node')
+
+const pipe = require('it-pipe')
+
+// Create a buffer and write the swarm key to it
+const swarmKey = Buffer.alloc(95)
+generate(swarmKey)
+
+// This key is for testing a different key not working
+const otherSwarmKey = Buffer.alloc(95)
+generate(otherSwarmKey)
+
+;(async () => {
+  const node1 = await privateLibp2pNode(swarmKey)
+  const node2 = await privateLibp2pNode(swarmKey)
+  // const node2 = await privateLibp2pNode(otherSwarmKey)
```

Comment:
@jacobheun I am having an issue using a invalid key. 

On the transport listener, for example `libp2p-tcp` https://github.com/libp2p/js-libp2p-tcp/blob/master/src/listener.js#L23 the `upgradeInbound` throws and the promise rejection is not handled. What do you recommend to do here? A `try...catch` in the encrypt and just log the error?

Reply:
btw, the error comes from:

https://github.com/libp2p/js-libp2p/blob/refactor/async-await/src/upgrader.js#L328
https://github.com/multiformats/js-multistream-select/blob/master/src/handle.js#L14
https://github.com/multiformats/js-multistream-select/blob/master/src/multistream.js#L37

Reply:
@vasco-santos the upgradeInbound should probably still throw. The Transport needs to know that there has been a failure so that it doesn't do any emitting. I think the best thing to do is to make sure that the transport listeners are catching that and logging errors. We should add a test for this to the transport interface and roll out patches to each of the transports.

Reply:
yes, I agree that should be the transports to handle this. Will PR everything